civis/utils/_jobs.py

```diff
@@ -30,3 +34,63 @@ def run_job(job_id, api_key=None, client=None):
                        (job_id, run['id']),
                        client=client,
                        poll_on_creation=False)
+
+
+def run_template(id, arguments, JSONValue=False, client=None):
+    """Run a template and return the results.
+
+    Parameters
+    ----------
+    id : int
+        The template id to be run.
+    arguments : dict
+        Dictionary of arguments to be passed to the template.
+    JSONValue : bool
+        If True, will return the JSON output of the template.
+        If False, will return the file ids associated with the
+        output results.
+    client : :class:`civis.APIClient`, optional
+        If not provided, an :class:`civis.APIClient` object will be
+        created from the :envvar:`CIVIS_API_KEY`.
+
+    Returns
+    -------
+    If JSONValue = False:
+      file_ids : dict
+        Dictionary of file ids with the keys being their output names.
+    If JSONValue = True:
+      json_output : dict
+        JSON dict containing the results of the template run.
+        Expects only a single JSON result. Will return nothing if
+        either there is no JSON result or there is more than 1 JSON result.
+
+    Examples
+    --------
+    >>> # Run template to return file_ids
+    >>> run_template(my_template_id, arguments=my_dict_of_args)
+    >>> # Run template to return JSON output
+    >>> run_template(my_template_id, arguments=my_dict_of_args, JSONValue=True)
+    """
+    if client is None:
+        client = APIClient()
+    job = client.scripts.post_custom(id, arguments=arguments)
+    run = client.scripts.post_custom_runs(job.id)
+    fut = CivisFuture(client.scripts.get_custom_runs, (job.id, run.id))
+    fut.result()
+    outputs = client.scripts.list_containers_runs_outputs(job.id, run.id)
+    if JSONValue:
+        json_output = [
+            o.value for o in outputs if o.object_type == "JSONValue"
+        ]
+        if len(json_output) == 0:
+            log.warning('No JSON output for template {}'.format(id))
+            return
```

Comment:
I don't too strongly about `None` vs `dict()`, let's move forward with `None` for now per @stephen-hoover's point.

Reply:
Sounds good.